tools/test-proxy/Azure.Sdk.Tools.TestProxy/Record.cs

```diff
@@ -48,8 +48,15 @@ public async Task Start()
         public void Stop([FromBody()] IDictionary<string, string> variables = null)
         {
             string id = RecordingHandler.GetHeader(Request, "x-recording-id");
+            bool save = true;
+            EntryRecordMode mode = RecordingHandler.GetRecordMode(Request);
 
-            _recordingHandler.StopRecording(id, variables: variables);
+            if (mode == EntryRecordMode.DontRecord)
```

Comment:
should we throw for other record modes since we are reusing the same enum?

Reply:
I see where you're coming from here. Will do.